subworkflows/nf-core/fastq_align_dedup_bismark/main.nf

```diff
@@ -0,0 +1,154 @@
+    ch_methylation_report         = Channel.empty()
+    ch_methylation_mbias          = Channel.empty()
+    ch_bismark_report             = Channel.empty()
+    ch_bismark_summary            = Channel.empty()
+    ch_multiqc_files              = Channel.empty()
+
+    /*
+     * Align with bismark
+     */
+    BISMARK_ALIGN (
+        ch_reads,
+        ch_fasta,
+        ch_bismark_index
+    )
+    ch_versions = ch_versions.mix(BISMARK_ALIGN.out.versions)
+
+    if (skip_deduplication) {
+    alignments        = BISMARK_ALIGN.out.bam
+    alignment_reports = BISMARK_ALIGN.out.report.map{ meta, report -> [ meta, report, [] ] }
+    } else {
+    /*
+    * Run deduplicate_bismark
+    */
+    BISMARK_DEDUPLICATE (
+        BISMARK_ALIGN.out.bam
+    )
+    alignments        = BISMARK_DEDUPLICATE.out.bam
+    alignment_reports = BISMARK_ALIGN.out.report.join(BISMARK_DEDUPLICATE.out.report)
+    ch_versions       = ch_versions.mix(BISMARK_DEDUPLICATE.out.versions)
+    }
+
+    /*
+     * Run bismark_methylation_extractor
+     */
+    BISMARK_METHYLATIONEXTRACTOR (
+        alignments,
+        ch_bismark_index
+    )
+    ch_methylation_bedgraph = BISMARK_METHYLATIONEXTRACTOR.out.bedgraph
+    ch_methylation_calls    = BISMARK_METHYLATIONEXTRACTOR.out.methylation_calls
+    ch_methylation_coverage = BISMARK_METHYLATIONEXTRACTOR.out.coverage
+    ch_methylation_report   = BISMARK_METHYLATIONEXTRACTOR.out.report
+    ch_methylation_mbias    = BISMARK_METHYLATIONEXTRACTOR.out.mbias
+    ch_versions             = ch_versions.mix(BISMARK_METHYLATIONEXTRACTOR.out.versions)
+
+    /*
+     * Run coverage2cytosine
+     */
+    if (cytosine_report) {
+    BISMARK_COVERAGE2CYTOSINE (
+        BISMARK_METHYLATIONEXTRACTOR.out.coverage,
+        ch_fasta,
+        ch_bismark_index
+    )
+    ch_coverage2cytosine_coverage = BISMARK_COVERAGE2CYTOSINE.out.coverage.collect{ it[1] }
+    ch_coverage2cytosine_report   = BISMARK_COVERAGE2CYTOSINE.out.report.collect{ it[1] }
+    ch_coverage2cytosine_summary  = BISMARK_COVERAGE2CYTOSINE.out.summary.collect{ it[1] }
+    ch_versions                   = ch_versions.mix(BISMARK_COVERAGE2CYTOSINE.out.versions)
+    }
+
+    /*
+     * Generate bismark sample reports
+     */
+    BISMARK_REPORT (
+        alignment_reports
+            .join(BISMARK_METHYLATIONEXTRACTOR.out.report)
+            .join(BISMARK_METHYLATIONEXTRACTOR.out.mbias)
+    )
+    ch_bismark_report = BISMARK_REPORT.out.report
+    ch_versions       = ch_versions.mix(BISMARK_REPORT.out.versions)
+
+    /*
+     * Generate bismark summary report
+     */
+    BISMARK_SUMMARY (
+        BISMARK_ALIGN.out.bam.collect{ it[1].name }.ifEmpty([]),
+        alignment_reports.collect{ it[1] }.ifEmpty([]),
+        alignment_reports.collect{ it[2] }.ifEmpty([]),
+        BISMARK_METHYLATIONEXTRACTOR.out.report.collect{ it[1] }.ifEmpty([]),
+        BISMARK_METHYLATIONEXTRACTOR.out.mbias.collect{ it[1] }.ifEmpty([])
```

Comment:
It would be nice to be more descriptive of what the `it` variable is. 
Does `toList` work here? Should this process still run even if all channels are empty()?

Reply:
@mahesh-panchal could you elaborate on 
> It would be nice to be more descriptive of what the it variable is.

need a comment beside on what the it variable for that line represents ?

Reply:
Rename the `it` variable to something e.g., `.collect { meta, bam -> bam }`

Reply:
thanks for catching this.....the process is actually not working as expected....fixing now